tuta-sdk/rust/src/id.rs

```diff
@@ -0,0 +1,70 @@
+use std::fmt::{Display, Formatter};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use serde::de::{Error, Visitor};
+
+#[derive(Clone, Default, PartialEq, PartialOrd, Debug)]
+#[repr(transparent)]
+pub struct Id(String);
+
+impl Id {
+    pub fn new(id: String) -> Self {
+        Self(id)
+    }
+
+    pub fn as_str(&self) -> &str {
+        &self.0
+    }
+
+    #[cfg(test)]
+    pub fn test_random() -> Self {
+        use base64::engine::Engine;
+
+        // not the actual alphabet we use in real generated IDs, but we aren't dealing with parsing generated IDs yet, so it's fine
+        let random_bytes: [u8; 9] = rand::random();
+        Self(base64::engine::general_purpose::URL_SAFE.encode(random_bytes))
+    }
+}
+
+impl From<Id> for String {
+    fn from(value: Id) -> Self {
+        value.0
+    }
+}
+
+impl Display for Id {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.write_str(self.as_str())
+    }
+}
```

Comment:
It could be worth considering [`derive_more`](https://crates.io/crates/derive_more) to avoid boilerplate when using the new type pattern.

Reply:
I think for display I would add that this is indeed an Id